python/servo/command_base.py

```diff
@@ -844,8 +858,15 @@ def configure_media_stack(self, media_stack: Optional[str]):
                 media_stack = "gstreamer"
             else:
                 media_stack = "dummy"
-        if media_stack != "dummy":
-            self.features += ["media-" + media_stack]
+
+        assert media_stack
+        media_stack = media_stack.lower()
+        if media_stack == "gstreamer":
+            self.media_stack = MediaStack.GSTREAMER
+        elif media_stack == "dummy":
+            self.media_stack = MediaStack.DUMMY
+        else:
+            raise RuntimeError(f"Wrong media stack {media_stack}")
```

Comment:
Just a small nit here to make this error a bit more comprehensible:

```suggestion
            raise RuntimeError(f"Got unknown media stack {media_stack}")
```

I would also move this up to the parsing of the configuration value so that we avoid converting the serialized configuration values to a string first. Essentially:

```python
    if self.config["build"]["media-stack"] != "auto":
        ...convert from string...
    elif (
                not self.cross_compile_target
                or ("armv7" in self.cross_compile_target and self.is_android_build)
                or "x86_64" in self.cross_compile_target
            ):
                media_stack = MediaStack.GSTREAMER
            else:
                media_stack = MediaStack.DUMMY
```

Reply:
I extracted this into static method, so it should be nicer now.